subsys/bluetooth/controller/ll_sw/nordic/lll/lll.c

```diff
@@ -785,43 +816,63 @@ static void preempt(void *param)
 	lll_prepare_cb_t resume_cb;
 	struct lll_event *next;
 	uint8_t idx;
-	int ret;
+	int err;
 
+	/* No event to abort */
 	if (!event.curr.abort_cb || !event.curr.param) {
 		return;
 	}
 
+	/* Check if any prepare in pipeline */
 	idx = UINT8_MAX;
 	next = ull_prepare_dequeue_iter(&idx);
 	if (!next) {
 		return;
 	}
 
+	/* Find a prepare that is ready and not a resume */
 	while (next && (next->is_aborted || next->is_resume)) {
 		next = ull_prepare_dequeue_iter(&idx);
 	}
 
+	/* No ready prepare */
 	if (!next) {
 		return;
 	}
 
-	ret = event.curr.is_abort_cb(next->prepare_param.param,
+	/* Preemptor not in pipeline */
+	if (next != param) {
+		uint32_t ret;
+
+		/* Start the preempt timeout */
+		ret = preempt_ticker_start(next, ticker_start_next_op_cb);
+		LL_ASSERT((ret == TICKER_STATUS_SUCCESS) ||
+			  (ret == TICKER_STATUS_BUSY));
```

Comment:
@mtpr-ot This is the change compare to your previous review. When a event in pipeline gets active before preempt timeout, then next event in pipeline shall setup new timeout. Otherwise, in the case of continuous scanning the window is not closed and pipeline overflows after a number of scan intervals as there is nothing to close the window, and done event to dequeue  the pipeline.